### PR TITLE
Various Documentation Improvements

### LINKED ISSUE #29: [BUG] Docstrings incomplete
**Describe the bug:**

The docstrings in `wrapper.py` are incomplete for the parameter structs.

**Additional context**

Especially the `AstroParams`, each should be defined in detail. Also `GlobalParams`.

### LINKED ISSUE #58: Wrong title for readthedocs
**Describe the bug:**

For 21cmFAST docs, the title is 21CMMC

image

### PR DESCRIPTION
**Pull request by steven-murray**
Closes #29 
Closes #58 

A number of improvements and checks for the docstrings throughout the code. Shouldn't change any functionality, but improves the Sphinx docs.

Adds checks for docstrings as well, so they don't get out of date.

### CODE REVIEW

**Comment by steven-murray** [comment]
@BradGreig there are a few parameters in the `GlobalParams` object that I didn't document because I wasn't sure I could do it well enough. They're in the `src/py21cmfast/inputs.py` module, and you can find them by searching for "???". Could you fill them out?

**Comment by BradGreig** [comment]
Hey @steven-murray, I have gone through and added in a brief description for the marked parameters. I also removed a parameter that was no longer used.

**Comment by steven-murray** [comment]
@BradGreig -- thanks a bunch!

**Comment by steven-murray** [comment]
@BradGreig this is now ready for a review -- all tests seem to be passing (as they should, there's little to no actual code changes in this PR).

File: src/py21cmfast/inputs.py [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+    wl : float
+        Dark energy equation of state parameter (wl = -1 for vacuum )
+    SHETH_b : float
+        Sheth-Tormen parameter for ellipsoidal collapse (for HMF).
+
+        .. note:: The best fit b and c ST params for these 3D realisations have a redshift,
+                  and a DELTA_R_FACTOR (see ANAL_PARAMS.H) dependence, as shown
+                  in Mesinger+. For converged mass functions at z~5-10, set `DELTA_R_FACTOR=1.1`
+                  and `SHETH_b=0.15` and `SHETH_c~0.05`.
+
```
Comment (BradGreig) [inline src/py21cmfast/inputs.py]: These refer to parameters located in the old 21cmFAST files. These should be updated to their new locations (i.e Globals.h and the globals struct).

File: docs/conf.py [hunk 2]
```diff
@@ -45,17 +49,17 @@ def __getattr__(cls, name):
...
 copyright = "{0}, {1}".format(year, author)
 version = release = "0.1.0"
 templates_path = ["templates"]
 
 pygments_style = "trac"
 extlinks = {
-    "issue": ("https://github.com/BradGreig/21CMMC/issues/%s", "#"),
-    "pr": ("https://github.com/BradGreig/21CMMC/pull/%s", "PR #"),
+    "issue": ("https://github.com/BradGreig/21cmFAST/issues/%s", "#"),
+    "pr": ("https://github.com/BradGreig/21cmFAST/pull/%s", "PR #"),
```
[the remark this answers is not in the document]
Comment (steven-murray) [reply docs/conf.py]: Good catch. Fixed.

File: docs/design.rst [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+=================
+The goal of v3 of ``21cmFAST`` is to provide the same computational efficiency and
+scientific features of the previous generations, but packaged in a form that adopts the
+best modern programming standards, including:
+
+* simple installation
+* comprehensive documentation
+* comprehensive test suite
+* more modular code
+* stadardised code formatting
```
[the remark this answers is not in the document]
Comment (steven-murray) [reply docs/design.rst]: Nice. Fixed.

File: src/py21cmfast/cli.py [hunk 4]
```diff
@@ -510,8 +591,28 @@ def coeval(ctx, redshift, config, regen, direc, z_step_factor, z_heat_max, seed)
 def lightcone(
     ctx, redshift, config, regen, direc, max_z, z_step_factor, z_heat_max, seed
 ):
-    """
-    Efficiently generate coeval cubes at a given redshift.
+    """Efficiently generate coeval cubes at a given redshift.
```
[the remark this answers is not in the document]
Comment (steven-murray) [reply src/py21cmfast/cli.py]: Fixed.

**Comment by steven-murray** [comment]
@qyx268 and I decided to get this (and some other structural PRs) merged before mini-halos in the interest of presenting a tutorial, so I'm going to merge now.

**Pull request merged**